client/asset/dcr/dcr.go

```diff
@@ -1629,7 +1629,43 @@ func (dcr *ExchangeWallet) LocktimeExpired(contract dex.Bytes) (bool, time.Time,
 		return false, time.Time{}, fmt.Errorf("error extracting contract locktime: %w", err)
 	}
 	contractExpiry := time.Unix(int64(locktime), 0).UTC()
-	return time.Now().UTC().After(contractExpiry), contractExpiry, nil
+	medianTime, err := dcr.calcPastMedianTime()
+	if err != nil {
+		return false, time.Time{}, fmt.Errorf("error calculating median time: %w", err)
+	}
+	return medianTime.After(contractExpiry), contractExpiry, nil
+}
+
+// calcPastMedianTime calculates the median time of the previous few blocks
+// prior to, and including, the best block.
+func (dcr *ExchangeWallet) calcPastMedianTime() (time.Time, error) {
+	dcr.tipMtx.RLock()
+	hash := dcr.currentTip.hash
+	dcr.tipMtx.RUnlock()
+
+	// Look at the last 11 blocks, which is consistant with dcrd.
```

Comment:
consistant

Reply:
consistent? Misspelling this for 20 years I guess.